marlowe/src/Language/Marlowe/Client.hs

```diff
@@ -0,0 +1,250 @@
+import qualified Data.Set                       as Set
+import qualified Language.PlutusTx              as PlutusTx
+import           Wallet                         ( WalletAPI(..)
+                                                , WalletAPIError
+                                                , intervalFrom
+                                                , throwOtherError
+                                                , createTxAndSubmit
+                                                , signature
+                                                , ownPubKeyTxOut
+                                                )
+import           Ledger                         ( DataScript(..)
+                                                , PubKey(..)
+                                                , Signature(..)
+                                                , Slot(..)
+                                                , TxOutRef
+                                                , TxIn
+                                                , TxOut
+                                                , TxOutOf(..)
+                                                , ValidatorScript(..)
+                                                , scriptTxIn
+                                                , scriptTxOut
+                                                )
+import qualified Ledger                         as Ledger
+import           Ledger.Validation
+import           Language.Marlowe.Common
+
+{- Mockchain instantiation of Marlowe Interpreter functions -}
+eqValue :: Value -> Value -> Bool
+eqValue = $$(equalValue)
+
+eqObservation :: Observation -> Observation -> Bool
+eqObservation = $$(equalObservation) eqValue
+
+eqContract :: Contract -> Contract -> Bool
+eqContract = $$(equalContract) eqValue eqObservation
+
+validContract :: State -> Contract -> Slot -> Ledger.Value -> Bool
+validContract = $$(validateContract)
+
+evalValue :: Slot -> [OracleValue Int] -> State -> Value -> Int
+evalValue pendingTxBlockHeight inputOracles = $$(evaluateValue) pendingTxBlockHeight inputOracles
+
+interpretObs :: [OracleValue Int] -> Int -> State -> Observation -> Bool
+interpretObs inputOracles blockNumber state obs = let
+    ev = evalValue (Slot blockNumber) inputOracles
+    in $$(interpretObservation) ev blockNumber state obs
+
+evalContract :: PubKey -> Input -> Slot
+    -> Ledger.Value -> Ledger.Value
+    -> State -> Contract
+    -> (State, Contract, Bool)
+evalContract = $$(evaluateContract)
+
+
+{-| Create Marlowe 'ValidatorScript' that remembers its owner.
+
+    At the end of a contract execution owner can spend an initial deposit
+    providing a 'Signature' for owner's public key.
+ -}
+marloweValidator :: PubKey -> ValidatorScript
+marloweValidator creator = ValidatorScript result where
+    result = Ledger.applyScript inner (Ledger.lifted creator)
+    inner  = Ledger.fromCompiledCode $$(PlutusTx.compile [|| $$(validatorScript) ||])
```

Comment:
There's now a helper function for the combination of `compile` and `fromCompiledCode`: `Ledger.compileScript`.